### ISSUE TITLE
Search provides odd results

### ISSUE THREAD

**Issue by jakobroehrl** [nextcloud/contacts#1176]
The complete vcard is searched not only the visible the UI contact fields.
That leads to odd search results if I search e.g. for a part of a telephonenumer 
When I download a vcard I find the telefonnumber part within the photo or in the UID

**Comment by skjnldsv** [comment]
You mean searching can also match the picture?
We indeed search though the whole vcard content, no matter what field it is.

I'm not sure I want to change it because this is quite simple and working for now, and adding more checks will decrease perfs.

Until we get #1158, where then it will be easier.

**Comment by jakobroehrl** [comment]
> You mean searching can also match the picture?
> We indeed search though the whole vcard content, no matter what field it is.

Yes, exactly.

**Comment by skjnldsv** [comment]
I'll leave this one open then, for searching purpose so that people do not create duplicate issues.

**Comment by georgehrke** [comment]
@skjnldsv Why not use address-book-queries?
https://tools.ietf.org/html/rfc6352#section-8.6.4

**Comment by skjnldsv** [comment]
@georgehrke because I'm not aware of all the dav rfc craziness? :grin:

**Comment by georgehrke** [comment]
> Instead of specifying al props one by one? This seems tedious!

No, as far as i know you have to declare them one by one.
But remember that there might be `X-PROPS`, so you don't want to search all anyway.

**Comment by skjnldsv** [comment]
@georgehrke seems like a very big request to do then :/
won't dav SEARCH also gives me what I want?
What do you recon ôh dav master! :bowing_woman: :grin:

**Comment by georgehrke** [comment]
What RFC are you referring to exactly when you say dav SEARCH?

**Comment by skjnldsv** [comment]
This one (I think?)
https://tools.ietf.org/html/rfc5323#section-2

**Comment by georgehrke** [comment]
I never tested that to be honest, but i would be extremely surprised if that's supported by our CalDAV and CardDAV server. 
These standards are meant to be used for "normal" "files"-WebDAV.

**Comment by skjnldsv** [comment]
@georgehrke and 6352 works already on carddav/caldav? :)

**Comment by georgehrke** [comment]
Yes, but unlike calendar-queries, which are implemented by ourselves for performance reasons, addressbook-queries are implemented by Sabre/DAV and it will be a bit more difficult to implement our own version.

So it will basically grabs all contacts from the database, parses them and see if they match. Not sure how good that scales.
